packages/@aws-cdk/aws-codedeploy/lib/lambda/custom-deployment-config.ts

```diff
@@ -0,0 +1,119 @@
+   */
+  LINEAR = 'Linear'
+}
+
+/**
+ * Properties of a reference to a CodeDeploy Lambda Deployment Configuration.
+ */
+export interface CustomLambdaDeploymentConfigProps {
+
+  /**
+   * The type of deployment config, either CANARY or LINEAR
+   */
+  readonly type: CustomLambdaDeploymentConfigType;
+
+  /**
+   * The integer percentage of traffic to shift:
+   * - For LINEAR, the percentage to shift every interval
+   * - For CANARY, the percentage to shift until the interval passes, before the full deployment
+   */
+  readonly percentage: string;
+
+  /**
+   * The interval, in number of minutes:
+   * - For LINEAR, how frequently additional traffic is shifted
+   * - For CANARY, how long to shift traffic before the full deployment
+   */
+  readonly interval: string;
+}
+
+/**
+ * A custom Deployment Configuration for a Lambda Deployment Group.
+ *
+ */
+export class CustomLambdaDeploymentConfig extends Construct implements ILambdaDeploymentConfig {
+
+  /**
+   * The name of the deployment config
+   */
+  public readonly deploymentConfigName: string;
+
+  /**
+   * The arn of the deployment config
+   */
+  public readonly deploymentConfigArn: string;
+
+  public constructor(scope: Construct, id: string, props: CustomLambdaDeploymentConfigProps) {
+    super(scope, id);
+    const deploymentType = 'TimeBased' + props.type.toString();
+    let routingConfig;
+    let customDeploymentConfigName;
+    if (props.type == CustomLambdaDeploymentConfigType.CANARY) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedCanary: {
+          canaryInterval: props.interval,
+          canaryPercentage: props.percentage,
+        },
+      };
+      customDeploymentConfigName = 'LambdaCanary' + props.percentage.toString() + 'Percent' + props.interval.toString() + 'Minutes';
+    } else if (props.type == CustomLambdaDeploymentConfigType.LINEAR) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedLinear: {
+          linearInterval: props.interval,
+          linearPercentage: props.percentage,
+        },
+      };
+      customDeploymentConfigName = 'LambdaLinear' + props.percentage.toString() + 'Percent' + props.interval.toString() + 'Minutes';
+    } else {
+      throw new Error('Deployment config type not recognized, only CANARY and LINEAR are supported');
+    }
+
+    // What the config is called in the AWS console
+    const resourceName = id + customDeploymentConfigName;
+
+    // The AWS Custom Resource that calls CodeDeploy to create and delete the resource
+    new AwsCustomResource(this, customDeploymentConfigName + 'DeploymentConfig', {
+      onCreate: {
+        service: 'CodeDeploy',
+        action: 'createDeploymentConfig',
+        parameters: {
+          deploymentConfigName: resourceName,
+          computePlatform: 'Lambda',
+          trafficRoutingConfig: routingConfig,
+        },
+        physicalResourceId: PhysicalResourceId.of(resourceName),
+      },
```

Comment:
Is `onUpdate` automatically taken care of by `AwsCustomResource` somehow? Or is the issue that you can't edit these after they've been created?

Reply:
onUpdate by default returns the same physical ID if nothing is specified for the onUpdate method. You can't edit these after configuration, it would be a delete and recreation.

Reply:
Added an onUpdate after some testing, now a user can change the construct in their code and easily get a new deployment configuration with different settings.